### ISSUE TITLE
Add a sub-project for map udf functions

### ISSUE THREAD

**Issue by zhuliquan** [apache/datafusion#11572]
### Describe the solution you'd like

I notice that code of `map`/`make_map` in sub-project `functions`. Beside, I notice that there is sub-project `functions-array` for array udf functions. Whether add a sub-project for map udf functions?

**Comment by goldmedal** [comment]
Actually, we're discussing moving the `map` related function to `functions-array` and renaming this crate to `functions-nested` in https://github.com/apache/datafusion/issues/11546#issuecomment-2241453265 because some map implementations are based on `make_array`. We can have a crate to collect all nested types.

**Comment by zhuliquan** [comment]
> Actually, we're discussing moving the `map` related function to `functions-array` and renaming this crate to `functions-nested` in [#11546 (comment)](https://github.com/apache/datafusion/issues/11546#issuecomment-2241453265) because some map implementations are based on `make_array`. We can have a crate to collect all nested types.

Ok, how long before we can see this crate (i.e. `function-nested`)? Which crate should contributors submit PR to if they adds UDFs about `map` now? Or do contributors have to wait until `funciotns-nested` is available to make corresponding contributions?

**Comment by goldmedal** [comment]
I plan to move the `map` UDF to `functions-array` in https://github.com/apache/datafusion/pull/11560 first, and then create another PR to rename `functions-array`. I guess it is just a renaming task. So, I think the new map-related functions can be put in `functions-array` currently. 
@jayzhan211 WDYT?

**Comment by jayzhan211** [comment]
> I plan to move the `map` UDF to `functions-array` in https://github.com/apache/datafusion/pull/11560 first, and then create another PR to rename `functions-array`. I guess it is just a renaming task. So, I think the new map-related functions can be put in `functions-array` currently. 
> @jayzhan211 WDYT?

Looks good to me

**Comment by zhuliquan** [comment]
> > I plan to move the `map` UDF to `functions-array` in #11560 first, and then create another PR to rename `functions-array`. I guess it is just a renaming task. So, I think the new map-related functions can be put in `functions-array` currently.
> > @jayzhan211 WDYT?
> 
> Looks good to me

Thank you, I got it
